aws-sns-subscription/src/main/java/software/amazon/sns/subscription/DeleteHandler.java

```diff
@@ -17,56 +22,39 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final UnsubscribeRequest unsubscribeRequest,
-        final ProxyClient<SnsClient> proxyClient) {
-
-        try {
-            logger.log(String.format("Deleting subscription for subscription arn: %s", unsubscribeRequest.subscriptionArn()));
-            proxyClient.injectCredentialsAndInvokeV2(unsubscribeRequest, proxyClient.client()::unsubscribe);
-        } catch (final SubscriptionLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final FilterPolicyLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final InvalidParameterException e) {
-            throw new CfnInvalidRequestException(e);
-        } catch (final InternalErrorException e) {
-            throw new CfnInternalFailureException(e);
-        } catch (final NotFoundException e) {
-            throw new CfnNotFoundException(e);
-        } catch (final AuthorizationErrorException e) {
-            throw new CfnAccessDeniedException(e);
-        } catch (final InvalidSecurityException e) {
-            throw new CfnInvalidCredentialsException(e);
-        } catch (final Exception e) {
-            throw new CfnInternalFailureException(e);
+        //Note that although we check the existence of Subscription ARN, it does not necessarily mean it's a valid one
+        //Subscription could be PendingConfirmation
+        if (resourceModel == null || com.amazonaws.util.StringUtils.isNullOrEmpty(subscriptionArn)) {
+            return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest, "Subscription ARN is required");
         }
 
-        logger.log(String.format("%s successfully deleted.", ResourceModel.IDENTIFIER_KEY_SUBSCRIPTIONARN));
-        return true;
+        final String topicArn = subscriptionArn.substring(0, subscriptionArn.lastIndexOf(ARN_SEPERATOR));
+        request.getDesiredResourceState().setTopicArn(topicArn);
+
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling Delete SNS Subscription", request.getStackId(),
+                request.getClientRequestToken()));
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+                .then(progress -> checkSubscriptionExistsAndSetSubscriptionArn(proxy, proxyClient, request, progress, callbackContext, true, logger))
+                .then(progress -> proxy.initiate("AWS-SNS-Subscription::Delete", proxyClient, resourceModel, callbackContext)
+                                    .translateToServiceRequest(Translator::translateToDeleteRequest)
+                                    .makeServiceCall((unsubscribeRequest, client) -> client.injectCredentialsAndInvokeV2(unsubscribeRequest, proxyClient.client()::unsubscribe))
+                                    .handleError((awsRequest, exception, client, model, context) -> handleError(awsRequest, exception, client, model, context))
```

Comment:
need to handle this error:

https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/subscription/SubscriptionActivitiesCustomImpl.java#:~:text=51%3A05%20PDT-,if%20(region%20%3D%3D%20null)%20%7B,%7D,-abc95fa8